### PR TITLE
feat: [CCM-7489]: Perspective Builder Enhancements

### PR DESCRIPTION
**Pull request by rajarshee-harness**
##### Summary:

- Replaced the value selector in perspective builder with tag input. (Now user can either select values from popover or type custom values).
- Create option shows in popover if there's no exact matching value for the search text in search results (Can click it to add a custom value).
- Removed snapshot tests.
- Also fixed a bug in cost categories where the create drawer was not opening when the list is empty.

Mocks - https://www.figma.com/file/QdoMvJqtuNnULhOiBsC3TV/Business-Mapping?node-id=758%3A16556

##### Jira Links:

https://harness.atlassian.net/browse/CCM-7489

##### Screenshots:

Screenshot 2022-05-09 at 11 47 15 AM

https://user-images.githubusercontent.com/97532027/167535410-7c992c02-a193-4c0b-bd4c-cc055bbbb55a.mov

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/75-ce/components/MultiValueSelectorComponent/PerspectiveBuilderMultiValueSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  const filteredValues = valueList.filter(val => val) as string[]
+
+  const [isCreateNewHighlighted, setIsCreateNewHighlighted] = useState(true)
+  const hoverProps = {
+    onMouseEnter: () => setIsCreateNewHighlighted(false),
+    onMouseLeave: () => setIsCreateNewHighlighted(true)
+  }
+
+  const renderValues = () => {
+    const allowToFetchMore = shouldFetchMore
```
Comment (rishik30) [inline src/modules/75-ce/components/MultiValueSelectorComponent/PerspectiveBuilderMultiValueSelector.tsx:48]: what's the need of duplicate variable when we already have the boolean value in props?

File: src/modules/75-ce/components/MultiValueSelectorComponent/PerspectiveBuilderMultiValueSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+          overscan={{ main: 20, reverse: 20 }}
+          endReached={
+            /* istanbul ignore next */ e => {
+              allowToFetchMore && fetchMore && fetchMore(e)
+            }
+          }
+          initialItemCount={filteredValues.length}
+          itemContent={(_, value) => {
+            const splitAt = `(${searchText})`
+            const re = new RegExp(splitAt, 'g')
```
Comment (rishik30) [inline src/modules/75-ce/components/MultiValueSelectorComponent/PerspectiveBuilderMultiValueSelector.tsx:64]: try to have some descriptive names for the variables.

File: src/modules/75-ce/components/MultiValueSelectorComponent/PerspectiveBuilderMultiValueSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        />
+      </Container>
+    )
+  }
+
+  const isSelectAllChecked = filteredValues.filter(val => selectedValues[val]).length === filteredValues.length
+
+  return (
+    <Container className={cx(css.valueContainer, css.createPerspectiveMvs)}>
+      {searchText && searchText?.length > 0 && !filteredValues.includes(searchText) ? (
```
Comment (rishik30) [inline src/modules/75-ce/components/MultiValueSelectorComponent/PerspectiveBuilderMultiValueSelector.tsx:118]: we can store the value of this check in a variable and use that as it's being used in couple of places.

File: src/modules/75-ce/components/MultiValueSelectorComponent/PerspectiveBuilderMultiValueSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+            onClick={() => createNewTag(searchText.split(/,\s*/))}
+          >
+            <Text inline color={Color.PRIMARY_7}>
+              {getString('create')}
+            </Text>
+            {searchText.split(/,\s*/).map(
+              (item, i) =>
+                item.length > 0 && (
+                  <Text
+                    key={i}
```
Comment (rishik30) [inline src/modules/75-ce/components/MultiValueSelectorComponent/PerspectiveBuilderMultiValueSelector.tsx:133]: have some unique key, no index in the key

**Comment by rajarshee-harness** [comment]
retrigger jest

File: src/modules/75-ce/components/PerspectiveBuilderFilters/views/OperandSelector.tsx [hunk 5]
```diff
@@ -265,7 +265,9 @@ const OperandSelector: React.FC<OperandSelectorProps> = ({
       }
     >
       <div className={css.operandSelectorContainer}>
-        {provider?.id && service?.id ? `${provider.name || provider.id} > ${service.name}` : 'Choose Operand'}
+        <Text lineClamp={1} font={{ variation: FontVariation.BODY2 }} color={Color.GREY_500}>
+          {provider?.id && service?.id ? `${provider.name || provider.id} > ${service.name}` : 'Choose Operand'}
```
Comment (rishik30) [inline src/modules/75-ce/components/PerspectiveBuilderFilters/views/OperandSelector.tsx:269]: put it in strings

**Review by rishik30 — CHANGES_REQUESTED** [review]

**Comment by rajarshee-harness** [comment]
retrigger jest

**Comment by rajarshee-harness** [comment]
retrigger jest

File: src/modules/75-ce/pages/business-mapping/BusinessMapping.tsx [hunk 6]
```diff
@@ -64,6 +64,33 @@ const BusinessMappingPage: () => React.ReactElement = () => {
     </Layout.Horizontal>
   )
 
+  const NewCostVategoryDrawer = (
```
Comment (jyotiarora-harness) [inline src/modules/75-ce/pages/business-mapping/BusinessMapping.tsx:67]: Correct the spelling `NewCostVategoryDrawer`, it should be `NewCostCategoryDrawer`

File: src/modules/75-ce/components/MultiValueSelectorComponent/PerspectiveBuilderMultiValueSelector.tsx [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+                  </Text>
+                )}
+              </Container>
+            )
+          }}
+          components={{
+            Footer: /* istanbul ignore next */ function renderFooter() {
+              return shouldFetchMore ? (
+                <Container padding="small" className={css.fetchingMoreLoader}>
+                  <Icon name="spinner" size={20} color="blue500" />
```
Comment (jyotiarora-harness) [inline src/modules/75-ce/components/MultiValueSelectorComponent/PerspectiveBuilderMultiValueSelector.tsx:103]: Don't use the color codes like this, use it like this `Color.BLUE_500`

**Comment by rajarshee-harness** [comment]
retrigger jest

**Review by rishik30 — APPROVED** [review]

**Review by jyotiarora-harness — APPROVED** [review]

**Pull request merged**